src/CommunityToolkit/Xamarin.CommunityToolkit/Views/HexLayout.shared.cs

```diff
@@ -0,0 +1,278 @@
+﻿using System;
+using Xamarin.Forms;
+
+namespace Xamarin.CommunityToolkit.UI.Views
+{
+    /// <summary>
+    /// A Layout that arranges the elements in a honeycomb pattern.
+    /// Based on https://github.com/AlexanderSharykin/HexGrid. Created by Alexander Sharykin.
+    /// </summary>
+    public class HexLayout : Layout<View>
+    {
+        public static readonly BindableProperty OrientationProperty =
+         BindableProperty.Create(nameof(Orientation), typeof(StackOrientation), typeof(HexLayout), StackOrientation.Vertical,
+             BindingMode.TwoWay, propertyChanged: (bindable, oldvalue, newvalue) => ((HexLayout)bindable).InvalidateMeasure());
+
+        public StackOrientation Orientation
+        {
+            get => (StackOrientation)GetValue(OrientationProperty);
+            set => SetValue(OrientationProperty, value);
+        }
+
+        public static readonly BindableProperty ColumnCountProperty =
+            BindableProperty.Create(nameof(ColumnCount), typeof(int), typeof(HexLayout), 1,
+                BindingMode.TwoWay, propertyChanged: (bindable, oldvalue, newvalue) => ((HexLayout)bindable).InvalidateMeasure());
+
+        public int ColumnCount
+        {
+            get => (int)GetValue(ColumnCountProperty);
+            set => SetValue(ColumnCountProperty, value);
+        }
+
+        public static readonly BindableProperty RowCountProperty =
+            BindableProperty.Create(nameof(RowCount), typeof(int), typeof(HexLayout), 1,
+                BindingMode.TwoWay, propertyChanged: (bindable, oldvalue, newvalue) => ((HexLayout)bindable).InvalidateMeasure());
+
+        public int RowCount
+        {
+            get => (int)GetValue(RowCountProperty);
+            set => SetValue(RowCountProperty, value);
+        }
+
+        public static readonly BindableProperty ColumnProperty =
+            BindableProperty.Create(nameof(Column), typeof(int), typeof(HexLayout), 1,
+                BindingMode.TwoWay, null);
+
+        public int Column
+        {
+            get => (int)GetValue(ColumnProperty);
+            set => SetValue(ColumnProperty, value);
+        }
+
+        int GetColumn(VisualElement e)
+        {
+            var column = (int)e.GetValue(ColumnProperty);
+            if (column >= ColumnCount)
+                column = ColumnCount - 1;
+            return column;
+        }
```

Comment:
```csharp
return Math.Min((int)e.GetValue(ColumnProperty), ColumnCount - 1);
```

Not sure that's my solution is better 🤓

Reply:
Mnn, I think is easier to read. Changed.